gcp/vm/data.tf

```diff
@@ -0,0 +1,11 @@
+resource "random_shuffle" "default" {
+  input        = ["${var.subnetworks}"]
+  result_count = 1
+}
+
+data "google_compute_zones" "available" {}

```

Comment:
Not sure about how GCP handles it, but IMHO it seems better to pass the `region` variable, to avoid having an available compute zone from other regions:
```hcl
data "google_compute_zones" "available" {
  region = "${var.region}"
}
```

Reply:
According to the [resource docs](https://www.terraform.io/docs/providers/google/d/google_compute_zones.html#region), region argument is optional and, in such case, region will be inferred from the provider. I think it would be helpful to declare region argument on the resource for the sake of readability but I'm not so sure about it. What you think?

Reply:
Ok, assuming that we'll have the region inferred from the provider, it should behave as expected. So, feel free to choose the best approach in your POV, setting this value can help with readability, you're right.